services/scribe/node/scribe.go

```diff
@@ -116,192 +116,173 @@ func (s Scribe) Start(ctx context.Context) error {
 	return nil
 }
 
-// TODO: Has issues with last confirmed data. Needs to be fixed.
+// confirmBlocks checks for reorgs with data stored in the scribe database
+// 0. Every few seconds (depending on ConfirmationRefreshRate in the config), the confirmBlocks function is called.
+// 1. First, the head block and the last confirmed block is retrieved. There is no "backfill" capability for reorgs.
+// it is suggested you reindex that range you want to confirm if it is far from the head.
+// 2. Block hashes for the blocks since the last confirmed block up to (latest block - the confirmation threshold)
+// are batch requested.
+// 3. These hashes are used to query the databases for receipts that do not have those block hashes.
+// 4. The returned receipts have their blocks deleted and re-backfilled.
+// 5. The entire range of blocks is then confirmed and last confirmed is updated.
 //
 //nolint:gocognit, cyclop
-func (s Scribe) confirmBlocks(ctx context.Context, chainID uint32, requiredConfirmations uint32) error {
-	logger.Infof("[LIVEFILL] start livefilling chain: %d", chainID)
-	newBlock, err := s.clients[chainID][0].BlockNumber(ctx)
+func (s Scribe) confirmBlocks(ctx context.Context, chainConfig config.ChainConfig) error {
+	chainID := chainConfig.ChainID
+	requiredConfirmations := chainConfig.ConfirmationConfig.RequiredConfirmations
+	getBlockBatchAmount := chainConfig.GetBlockBatchAmount
+	if getBlockBatchAmount == 0 {
+		getBlockBatchAmount = 25
+	}
+	confirmationThreshold := chainConfig.ConfirmationConfig.ConfirmationThreshold
+
+	latestBlock, err := s.clients[chainID][0].BlockNumber(ctx)
```

Comment:
Essentially, blocks between the last confirmed block (1000 below the head if no last confirmed block) and latest block - the confirmation threshold will be checked.